src/models/trainerMatchingDao.js

```diff
@@ -0,0 +1,190 @@
+        u.gender AS gender,
+        t.specialized AS specialized,
+        u.img_url AS imgUrl
+        FROM products p 
+        JOIN trainers t ON t.id = p.trainer_id
+        JOIN users u ON u.id = t.user_id
+        WHERE p.status = 1
+        ${categoryQuery}
+        ${genderQuery}
+        ${trainerCheckQuery}
+        ${sortQuery}
+        ${products};
+        `);
+  return result;
+};
+
+//트레이너 상세 정보
+const getTrainerMatchingDetail = async (userId, productsId) => {
+  const [result] = await AppDataSource.query(
+    `
+        SELECT 
+        p.id AS id,
+        p.trainer_id AS trainerId,
+        p.available_area AS availableArea,
+        p.available_time AS availableTime,
+        p.category_name AS categoryName,
+        p.term AS term,
+        p.price AS price,
+        p.content AS content,
+        p.created_at AS createdAt ,
+        u.gender AS gender,
+        t.specialized AS specialized,
+        u.img_url AS imgUrl
+        FROM products p 
+        JOIN trainers t ON t.id = p.trainer_id
+        JOIN users u ON u.id = t.user_id
+        WHERE p.status = 1
+        AND u.id = ?
+        AND p.id = ?;
+        `,
+    [userId, productsId]
+  );
+  return result;
+};
+
+const isTrainer = async (userId) => {
+  const [result] = await AppDataSource.query(
+    `SELECT CASE
+        WHEN (SELECT u.nickname FROM trainers t
+            JOIN users u ON t.user_id  = u.id
+            WHERE u.id = ?) IS NULL
+            THEN 'false'
+            ELSE 'true'
+        END AS isTrainer;
+    `,
+    [userId]
+  );
+  return result.isTrainer;
+};
+// 구독자 여부 확인
+const isSubscribed = async (userId) => {
+  const [result] = await AppDataSource.query(
+    `SELECT CASE
+        WHEN (SELECT u.nickname FROM sub_orders so
+            JOIN users u ON so.user_id  = u.id
+            WHERE u.id = ?) IS NULL
+            THEN 'false'
+            ELSE 'true'
+        END AS isSubscribed;
+      `,
+    [userId]
+  );
+  return result.isSubscribed;
+};
+const isPostedTrainer = async (productsId, trainerId) => {
+  const [result] = await AppDataSource.query(
+    `SELECT CASE
+        WHEN (SELECT p.id FROM p
+            WHERE p.id = ?
+            AND p.trainer_id = ?) IS NULL
+            THEN 'false'
+            ELSE 'true'
+        END AS isPostedTrainer;
+          `,
+    [productsId, trainerId]
+  );
+  return result.isPostedTrainer;
+};
+
+const findTrainerId = async (userId) => {
+  const [trainerId] = await AppDataSource.query(
+    `
+    SELECT t.id AS id
+    FROM trainer t
+    JOIN users u ON u.id = t.user_id
+    WHERE u.id = ?;`,
+    [userId]
+  );
+  return trainerId.id;
+};
+const findCategoryName = async (trainerId) => {
+  const [category] = await AppDataSource.query(
+    `
+    SELECT wc.category AS name
+    FROM workout_category wc
+    JOIN trainers t ON t.specialized = wc.id
+    WHERE t.id = ?;`,
+    [trainerId]
+  );
+  return category.name;
+};
+const postTrainerMatching = async (
```

Comment:
```suggestion
const createTrainerMatching = async (
```

1. POST는 데이터를 입력한다는 의미가 있지만, 이는 HTTP method의 경우에 적용되는 명명 규칙이어서, Dao layer에서 사용하실 경우 create, get(find), update, delete 가운데 함수 이름을 선정하시는 편이 좋습니다.
2. 해당 함수는 두 개 이상의 테이블에 변동을 일으키는 작업을 포함하고 있기 때문에 transaction 걸어주시는 것이 좋겠습니다 :)

Reply:
넵 감사합니다 확인할게요!